### ISSUE TITLE
Render/Zoom/Refresh ordering

### ISSUE THREAD

**Issue by andrew-davie** [osmandapp/Osmand#2408]
I think there's a small bug in the process of rendering/zooming.
What I think is happening is that there are a number of "zoom" levels which require complete redraws/renders because, for example, the tile sets are completely different. These are powers of two different from each other. Between those levels are "scaled" images of the lower level. So when you zoom in, you see alternating next zoom level, or a scale of the existing level. That's all fine.
But if you look at the zooming in this video:  https://youtu.be/WkT1JFqYjPc
At around 26 seconds I zoom in and the arrows become visible for the first time. If you look carefully you see them briefly "big" arrows, and then back to the correct size. What I think is happening here is that as I zoom in, the scaler first kicks in to show the current data at the higher zoom scale (that's OK) and then the arrows are drawn, and *then the screen is refreshed* and THEN the zoom scaler is reset, and *then the screen is refreshed* so we see the new zoom level at the unscaled version.  This is wrong, because it shows the arrows (and anything else) briefly at a magnified version unsuitable for the zoom for which it was drawn. The correct solution, I think, is to reset the zoom scaler just before the first refresh.
I hope this makes sense and you can understand what I'm trying to explain.
TL;DR - scaling should not flash big and then revert to correct size.

**Comment by vshcherb** [comment]
The way it is implemented gives pretty good FPS because it is scaling in a different thread. Of course in your case you have pretty complex code running in the 2nd thread, FPS of this thread should be around 5-7 frame per seconds, FPS of real UI thread 40-50 FPS.
So technically the issue could be solved only if everything would work in 1 thread and this is far from possible today.
